### ISSUE TITLE
dynamo: mutation to a LOAD_BUILD_CLASS-created class's sourceless attribute is lost across a graph break

### ISSUE THREAD

**Issue by aperson30** [pytorch/pytorch#192872]
### 🐛 Describe the bug

```python
import torch
import torch._dynamo

torch._dynamo.config.enable_trace_load_build_class = True


def fn(t):
    class Holder:
        items = []

    Holder.items.append(1)
    y = t * 2
    torch._dynamo.graph_break()
    Holder.items.append(2)
    z = y * 3
    return z.sum() + len(Holder.items)


t = torch.randn(3)
opt_fn = torch.compile(fn, backend="eager")
out = opt_fn(t)
expected = fn(t)
print("compiled:", out.item())
print("eager:   ", expected.item())
assert torch.equal(out, expected), "mismatch"
```

Expected: compiled result matches eager. Actual: compiled result is off by
exactly one element of `Holder.items` -- the append before the graph break
is dropped; only the second append survives.

Found while working on #151251 / #192272, which fixes mutation-loss for a
mutable sourceless class attribute *within* a single compiled region (a
class with no source, e.g. one built via `LOAD_BUILD_CLASS`, previously got
a fresh `VariableTracker` on every attribute read, silently dropping
mutations made through an earlier read). That fix -- and a follow-up
redesign routing it through the same `id_to_variable`/`AttributeMutationNew`
identity-tracking machinery Dynamo already uses for other sourceless
objects (e.g. Enum/DispatchKey members in `SourcelessBuilder`) -- does not
address values crossing a graph-break/resume-function boundary. The resume
function appears to reconstruct the attribute's contents from only what it
observes in its own trace, not the real object's already-mutated state at
that point.

`torch._dynamo.config.enable_trace_load_build_class` must be `True` to hit
this (`LOAD_BUILD_CLASS` tracing is opt-in).

### Versions

Reproduced on a local build from pytorch/pytorch main (dynamo, CPU-only,
backend="eager"), same result on the fix in #192272 both before and after
the redesign commit -- confirms this is pre-existing, not a regression from
that PR.

cc @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @kadeng @amjames @jataylo @azahed98 @jansel

**Comment by guilhermeleobas** [comment]
> torch._dynamo.config.enable_trace_load_build_class must be True to hit
this (LOAD_BUILD_CLASS tracing is opt-in).

FYI, this flag is disabled by default as support LOAD_BUILD_CLASS is not a trivial task. We only enabled it in some tests (CPython tests) to avoid a graph break there.
